imagebuildah/stage_executor.go

```diff
@@ -1216,6 +1259,18 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 			}
 		}
 
+		// Small wrapper around s.pushCache to prevent duplication of code
+		pushCache := func() error {
```

Comment:
Is this necessary? I was expecting something like
```go
if cacheID == "" || !cachePulled {
   // body of this helper inlined
}
```
after the `if cacheID != ""` below, directly inline in the per-stage flow. (That would also make it a tiny bit easier to see under which conditions cache images are pushed, without having to trace the execution flow at all.)

Reply:
moved to below line.